### DISCUSSION TITLE
Is it possible to enable hardware support on iOS?

### DISCUSSION THREAD

**Discussion by Yves0** [openssl/openssl#26008] [Q&A] [answered]
we followed the [wiki](https://wiki.openssl.org/index.php/Compilation_and_Installation#64-bit) to build openssl for iOS.And I notice a config `no-hw` there. So can we just remove this config and enable hardware support?

**Comment by Sashan** [comment, marked as answer]
I assume you are trying to build openssl-3.4 or master branch as found on github.

if I understand things right the `hw-` options enables disables HW components such as _padlock_ (`hw-padlock` vs. `no-hw-padlock`). So it should be OK to omit the option completely.

perhaps you are rather interested in optimized implementations for various cryptographic algorithms such as aes, sha*, etc. these are enabled by default. those optimized implementations can be disabled by adding `no-asm` compile time option. If `no-asm` option is missing then assembly code is enabled for compilation.

**Reply by Yves0** [reply]
Thank you! Yes, I’m looking for a hardware-accelerated AES algorithm implementation for a C/C++ library on iOS and Android.
Is it enabled by default in OpenSSL 1.1.x as well?

**Reply by Sashan** [reply]
Yes, OpenSSL 1.1.x should enable optimized assembly implementations by default too.
